prow/test/integration/fakegerritserver/fakegerritserver.go

```diff
@@ -98,21 +121,172 @@ func defaultHandler() func(*http.Request) (interface{}, int, error) {
 	}
 }
 
-func changesHandler() func(*http.Request) (interface{}, int, error) {
+// GetBranch
+func projectHandler(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
+	return func(r *http.Request) (interface{}, int, error) {
+		vars := mux.Vars(r)
+		projectName := vars["project-name"]
+		branchID := vars["branch-id"]
+		if res := fgc.GetBranch(projectName, branchID); res != nil {
+			content, err := json.Marshal(res)
+			if err != nil {
+				return "", http.StatusInternalServerError, err
+			}
+			return string(content), http.StatusOK, nil
+		}
+		return "", http.StatusNotFound, nil
+	}
+}
+
+// Admin endpoint to add a change to the Fake Gerrit Server
+func addHandler(fgc *fakegerrit.FakeGerrit) func(*http.Request) (interface{}, int, error) {
+	return func(r *http.Request) (interface{}, int, error) {
+		change := gerrit.ChangeInfo{}
+		err := unmarshal(r, &change)
+		if err != nil {
+			logrus.Infof("Error unmarshling: %v", err)
+			return "", http.StatusInternalServerError, err
+		}
+		fgc.AddChange("fakegerritserver", &change)
+
+		logrus.Infof("The Entire Fgc: %v", fgc)
```

Comment:
Do we really need to log the entire state here?

Reply:
Forgot to delete this when I was debugging.

Reply:
Forgot to delete this when I was debugging.